File: app/src/main/java/zechs/zplex/utils/LXStringRequest.java
package zechs.zplex.utils;


import com.android.volley.NetworkResponse;
import com.android.volley.Response;
import com.android.volley.toolbox.HttpHeaderParser;
import com.android.volley.toolbox.StringRequest;

import java.nio.charset.StandardCharsets;

public class LXStringRequest extends StringRequest {

    public LXStringRequest(int method, String url, Response.Listener<String> listener, Response.ErrorListener errorListener) {
        super(method, url, listener, errorListener);
    }

    @Override
    protected Response<String> parseNetworkResponse(NetworkResponse response) {
        return Response.success(new String(response.data, StandardCharsets.UTF_8), HttpHeaderParser.parseCacheHeaders(response));
    }
}

File: app/src/main/java/zechs/zplex/utils/LXStringRequest.kt
package zechs.zplex.utils

import com.android.volley.NetworkResponse
import com.android.volley.Response
import com.android.volley.toolbox.HttpHeaderParser
import com.android.volley.toolbox.StringRequest
import java.nio.charset.StandardCharsets

class LXStringRequest(
    method: Int,
    url: String?,
    listener: Response.Listener<String?>?,
    errorListener: Response.ErrorListener?
) : StringRequest(method, url, listener, errorListener) {
    override fun parseNetworkResponse(response: NetworkResponse): Response<String> {
        return Response.success(
            String(response.data, StandardCharsets.UTF_8),
            HttpHeaderParser.parseCacheHeaders(response)
        )
    }
}